### ISSUE TITLE
Feature engineering - Don't generate code for unselected columns

### ISSUE THREAD

**Issue by rustd** [dotnet/machinelearning-modelbuilder#242]
If you deselect columns in the feature engineering step, then the generated code still refers to the columns. 

This is confusing from the user perspective that they will see columns in ModelInput.cs which they ignored before. ModelInput.cs is the class users will use more in their projects and they will end up providing values for these ignored columns.  
Also in the generated code in Program.cs we are still printing the columns the user had dropped. 

![image](https://user-images.githubusercontent.com/1132652/65454778-b9c88700-ddfa-11e9-9cd9-cdea99fc75e9.png)

**Comment by JakeRadMSFT** [comment]
Closing issue and will wait for customer feedback.

**Comment by EchoNBounce** [comment]
I'm a 'customer' and I agree, this behaviour baffled me.

For one it undermines confidence that the trained model is actually based on just the columns you specified at training time. I ended up running training several times including creating new versions of the source file with columns excluded just to make sure something wasn't going wrong.

And, as @rustd said, it means class consumers don't actually know what they _need_ to provide to yield a prediction. I think most people would read a name like ModelInput and assume it is the parsimonious set of required parameters.

**Comment by Ki6FHX** [comment]
Customer. I have a column in the training data that I don't want to be used. I deselected it. However, I get this error:
"Specified column full_status is not found in the dataset." I deselected "fullstatus" column and don't want it to train off this column.

Stack:
"at Microsoft.ML.AutoML.ColumnInferenceValidationUtil.ValidateSpecifiedColumnsExist(ColumnInformation columnInfo, IDataView dataView)
   at Microsoft.ML.AutoML.ColumnInferenceApi.InferColumns(MLContext context, String path, ColumnInformation columnInfo, Boolean hasHeader, ColumnSplitResult splitInference, InferenceResult typeInference, Boolean trimWhitespace, Boolean groupColumns)
   at Microsoft.ML.AutoML.ColumnInferenceApi.InferColumns(MLContext context, String path, ColumnInformation columnInfo, Nullable`1 separatorChar, Nullable`1 allowQuotedStrings, Nullable`1 supportSparse, Boolean trimWhitespace, Boolean groupColumns, Boolean hasHeader)
   at Microsoft.ML.ModelBuilder.AutoMLEngine.InferColumns(MLContext context, AutoMLServiceParamater config, ColumnInformation columnInformation)
   at Microsoft.ML.ModelBuilder.AutoMLEngine.<StartTrainingAsync>d__30.MoveNext() in /_/src/Microsoft.ML.ModelBuilder.AutoMLService/AutoMLEngineService/AutoMLEngine.cs:line 112"
